http-body-util/src/either.rs

```diff
@@ -0,0 +1,196 @@
+use std::error::Error;
+use std::fmt::Debug;
+use std::pin::Pin;
+use std::task::{Context, Poll};
+
+use bytes::Buf;
+use http::HeaderMap;
+use http_body::{Body, SizeHint};
+use proj::EitherProj;
+
+/// Sum type with two cases: `Left` and `Right`, used if a body can be one of
+/// two distinct types.
+#[derive(Debug, Clone, Copy)]
+pub enum Either<L, R> {
+    /// A value of type `L`
+    Left(L),
+    /// A value of type `R`
+    Right(R),
+}
+
+impl<L, R> Either<L, R> {
+    /// This function is part of the generated code from `pin-project-lite`,
+    /// for a more in depth explanation and the rest of the generated code refer
+    /// to the [`proj`] module.
+    pub(crate) fn project(self: Pin<&mut Self>) -> EitherProj<L, R> {
+        unsafe {
+            match self.get_unchecked_mut() {
+                Self::Left(left) => EitherProj::Left(Pin::new_unchecked(left)),
+                Self::Right(right) => EitherProj::Right(Pin::new_unchecked(right)),
+            }
+        }
+    }
+}
+
+impl<L> Either<L, L> {
+    /// Convert [`Either`] into the inner type, if both `Left` and `Right` are
+    /// of the same type.
+    pub fn into_inner(self) -> L {
+        match self {
+            Either::Left(left) => left,
+            Either::Right(right) => right,
+        }
+    }
+}
+
+impl<L> Into<L> for Either<L, L> {
+    fn into(self) -> L {
+        self.into_inner()
+    }
+}
```

Comment:
Do we need these conversions? The enum variants are public so a user can break it down. The main thing they need is just the body impl. I also believe this is failing CI.

Reply:
I must've forgotten to run the tests on my machine before pushing the `Into<L>` implementation 🤦 I don't think `Into<L>` is needed. Still, I believe converting into the inner type is common enough to warrant a dedicated method.